src/overlaybd/registryfs/registryfs.cpp

```diff
@@ -114,16 +114,21 @@ class RegistryFSImpl : public RegistryFS {
     long GET(const char *url, photon::net::HeaderMap *headers, off_t offset, size_t count,
              photon::net::IOVWriter *writer, uint64_t timeout) {
         Timeout tmo(timeout);
-        long ret = 0;
-        UrlInfo *actual_info = m_url_info.acquire(url, [&]() -> UrlInfo * {
-            return getActualUrl(url, tmo.timeout(), ret);
-        });
-        if (actual_info == nullptr)
-            return ret;
+        const char *actual_url = url;        
 
-        const char *actual_url = url;
-        if (actual_info->mode == UrlMode::Redirect)
-            actual_url = actual_info->info.data();
+        // skip resolving source url if p2p is enabled and ignoreRedirect is set to true
+        if !(m_accelerate.size() > 0 && m_ignoreRedirect) {
+            long ret = 0;
+            UrlInfo *actual_info = m_url_info.acquire(url, [&]() -> UrlInfo * {
+                return getActualUrl(url, tmo.timeout(), ret);
+            });
+            if (actual_info == nullptr)
+                return ret;
+
+            if (actual_info->mode == UrlMode::Redirect)
+                actual_url = actual_info->info.data();
+        }
+        
```

Comment:
getActualUrl() will get the 'Bearer Token' and put it to HTTP Header, see 
https://github.com/containerd/overlaybd/blob/main/src/overlaybd/registryfs/registryfs.cpp#L145


If ignoreRedirect is true, 
overlaybd will send the request URL to Kraken listened port without any authentication ? is this what you want?